src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -56,51 +112,7 @@ export const useSwapper = () => {
     TradeAsset | undefined,
     Trade<KnownChainIds>,
   ]
-  const adapterManager = useChainAdapters()
-  const [swapperManager] = useState<SwapperManager>(() => new SwapperManager())
-
-  useEffect(() => {
-    if (!adapterManager || !swapperManager) return
-
-    const web3 = getWeb3Instance()
-
-    // TODO: Uncomment when we are ready for a Thorchain swapper
-    // ;(async () => {
-    //   const midgardUrl = getConfig().REACT_APP_MIDGARD_URL
-    //   const thorSwapper = new ThorchainSwapper({
-    //     midgardUrl,
-    //     adapterManager,
-    //     web3,
-    //   })
-    //   await thorSwapper.initialize()
-    //   swapperManager.addSwapper(thorSwapper)
-    // })()
-
-    const ethereumChainAdapter = adapterManager.get(
-      KnownChainIds.EthereumMainnet,
-    ) as unknown as ethereum.ChainAdapter
-
-    const zrxEthereumSwapper = new ZrxSwapper({
-      web3,
-      adapter: ethereumChainAdapter,
-    })
-
-    const avalancheChainAdapter = adapterManager.get(
-      KnownChainIds.EthereumMainnet,
-    ) as unknown as ethereum.ChainAdapter
-
-    const zrxAvalancheSwapper = new ZrxSwapper({
-      web3,
-      adapter: avalancheChainAdapter,
-    })
-
-    try {
-      swapperManager.addSwapper(zrxEthereumSwapper)
-      swapperManager.addSwapper(zrxAvalancheSwapper)
-    } catch (e) {
-      moduleLogger.error(e, { fn: 'addSwapper' }, 'error adding swapper')
-    }
-  }, [adapterManager, swapperManager])
+  const [swapperManager] = useState<SwapperManager>(getSwapperManager())
```

Comment:
Feels weird only getting the state field and not the setter but I get the idea on how we want to be reactive to the `Maybe<SwapperManager>` unwrapping, and I don't really see any other way to achieve this as long as this is a hook so 🍬

Reply:
@elmutt and i gave this a try yeah, using `getSwapperManager()` everywhere works and has the same effect, but stylistically looks odd.

it's not actually a `Maybe` - it's concrete but w/ side effects